### ISSUE TITLE
Install for newbies

### ISSUE THREAD

**Issue by juanMarinero** [neoclide/coc.nvim#2888]
The _coc.nvim_ [wiki](https://github.com/neoclide/coc.nvim/wiki) documentation is excellent!!  Actually this issue was inspired by its [quick-start](https://github.com/neoclide/coc.nvim#quick-start). But for **new Vim-users** it might be overwhelming.
Same **learning wall** affects many other _Vim_'s basic toolkit plugins ([ALE](https://github.com/dense-analysis/ale), [vimspector](https://github.com/puremourning/vimspector), [fzf](https://github.com/junegunn/fzf.vim), ...)

Next text aims to enable [coc.nvim](https://github.com/neoclide/coc.nvim) plugin the easiest way for Vim beginners. Because of this mindset, it deals not with:
 - multiple install-/customization ways: like edit [configuration files](https://github.com/neoclide/coc.nvim/wiki/Using-the-configuration-file) or further [plugins](https://github.com/neoclide/coc.nvim/wiki/Using-coc-extensions#use-vims-plugin-manager-for-coc-extension). Just `:CocInstall [cmd]` (if possible).
 - mutiple OS (just _Ubuntu_), 
 - _NeoVim_ (just _Vim_), 
 - multiple Vim-plugin-managers (just [vim-plug](https://github.com/junegunn/vim-plug)). For _Vundle_, _NeoBundle_ or _Pathogen_ [see me](https://vimawesome.com/plugin/coc-nvim).

Basic rule (for everything in life): implement minimal change, test it, carry on.
And avoid all potential conflicts. Till _coc.nvim_ works, uninstall other plugins: in `~/.vimrc` comment all in-between `call plug#begin('~/.vim/plugged')` and `call plug#end()`, then `:source %` and `:PlugClean`. Embrace the [MWE](https://stackoverflow.com/help/minimal-reproducible-example) mentality.

### TL;DR

1. Install requirements ([wiki](https://github.com/neoclide/coc.nvim/wiki/Install-coc.nvim)): newest [Vim](https://github.com/vim/vim/blob/master/src/INSTALL) and _NodeJS_. 
As well as _python_ virtual environment (for _py_ files), [LaTeX](https://www.latex-project.org/get/#the-latex-git-repository) (for _tex_ files), ...

2. Install _coc.nvim_: 
    `$ vim ~/.vimrc` and there run `:execute "normal /call plug#end()\<CR>OPlug 'neoclide/coc.nvim', {'branch': 'release'}" | w | source $MYVIMRC"`.
    Run and wait `:PlugInstall`
    Check it works, run `:CocInfo`

3. Add _coc.nvim_'s shorcuts to `~/.vimrc`, minimun [these](https://github.com/neoclide/coc.nvim#example-vim-configuration) or in [detail](https://github.com/neoclide/coc.nvim/wiki/Completion-with-sources#use-cr-to-confirm-completion):
    ```
    nmap <silent> <leader>gd <Plug>(coc-definition)
    nmap <silent> <leader>gr <Plug>(coc-references)
    inoremap <silent><expr> <c-@> coc#refresh()
    nmap <leader>rn <Plug>(coc-rename)
    ```

4. Install language specific:
4.1. search language in [here](https://github.com/neoclide/coc.nvim/wiki/Language-servers#contents) or [here](https://github.com/neoclide/coc.nvim/wiki/Using-coc-extensions#implemented-coc-extensions).
4.2 Among installation options (servers, plugins, edit config-file), pick one with `:CocInstall [whatever]` (if available for that language). E.g. for _javascript_ ([see](https://github.com/neoclide/coc.nvim/wiki/Language-servers#javascripttypescript)) run `:CocInstall coc-tsserver`.
Note each might need previously to install further requirements. Or settings. Or _vimscript_ conflicts,... read "Errors" section below.
4.3  test it, e.g.: 
- `:tabnew deleteme.js`. Populate it: `:execute 'normal Gofunction f01(x){ console.log("x: "+ x)} f01("x_value");'`
- find call of `f01` (just run `:execute "normal G$?f01\<CR>"` )
- and check that `<leader>gd` moves cursor to function declaration.

### Errors
- **must read**: _coc.nvim_'s [FAQs](https://github.com/neoclide/coc.nvim/wiki/F.A.Q) and [trouble-shooting](https://github.com/neoclide/coc.nvim#trouble-shooting)
- search in these order:
 1. _coc.nvim_ [wiki](https://github.com/neoclide/coc.nvim/wiki)
 2. _coc.nvim_ [issues](https://github.com/neoclide/coc.nvim/issues)
 3. language server specific issues (like [this](https://github.com/pappasam/coc-jedi/issues))
 4.  _google_ or [aike](https://duckduckgo.com/), [vim-docu](https://vimhelp.org/), [reddit](https://www.reddit.com/r/vim/), [stackexchange](https://vi.stackexchange.com/), [stackoverflow](https://stackoverflow.com/questions/tagged/vim),...
- if issues with other Vim-plugin,... Are you sure that's the cause? Is there same error if you `:PlugClean` that [or all] plugin?
- issues with other Vim-plugins might not be trivial. Luckily common conflicts are well documented. E.g. using [jedi-vim](https://github.com/davidhalter/jedi-vim) plugin, install [coc-jedi ](https://github.com/neoclide/coc.nvim/wiki/Language-servers#python) and correct `~/.vimrc` settings ([link](https://github.com/neoclide/coc-python/issues/220#issuecomment-673689526))
- maybe language server installation guide lacks steps/requirements. E.g. _coc-clangd_ does not specify to extend `$PATH` env variable ([coc-clangd issue](https://github.com/clangd/coc-clangd/issues/61#issuecomment-777139827)).
- maybe you tried too fast, not even reading error messages and not installing obvious language requirements ([shame on me](https://github.com/pappasam/coc-jedi/issues/25) 😅) 
- Or maybe these basic steps work, but you have problems customizing the [configuration files](https://github.com/neoclide/coc.nvim/wiki/Using-the-configuration-file). Then, present text is not aimed for you, sorry.
- If need help: anywhere you ask, [ask a good question](https://www.reddit.com/r/vim/comments/e7ybgs/how_to_ask_good_questions_on_rvim/), specially embrace the [MWE](https://stackoverflow.com/help/minimal-reproducible-example) mentality. In GitHub a bug-report already helps with section and comments ([see](https://github.com/neoclide/coc.nvim/issues/new?assignees=&labels=&template=bug_report.md)) like `## Result from CocInfo`, `## Describe the bug`, `## Reproduce the bug`,... see a real example in [#2670](https://github.com/neoclide/coc.nvim/issues/2670). BUT since if you are reading this, you are likely a newbie (like me, no worry, little by little is still progress), then I doubt your bug was not already reported, and it's more probably a personal issue (like skipping an installation step or incompatible `~/.vimrc` lines), therefore, **avoid** at all cost **open a new GitHub issue**. If struggle post in [reddit](https://www.reddit.com/r/vim/), [stackexchange](https://vi.stackexchange.com/), [stackoverflow](https://stackoverflow.com/questions/tagged/vim),... if more advance users say it's a bug, then yes report in it GitHub.
